stdlib/random.mli

```diff
@@ -18,18 +18,18 @@
 (** {1 Basic functions} *)
 
 val init : int -> unit
-(** Initialize the generator, using the argument as a seed.
-     The same seed will always yield the same sequence of numbers. *)
+(** Initialize the domain-local generator, using the argument as a seed.
+    The same seed will always yield the same sequence of numbers. *)
 
 val full_init : int array -> unit
 (** Same as {!Random.init} but takes more data as seed. *)
 
 val self_init : unit -> unit
-(** Initialize the generator with a random seed chosen
-   in a system-dependent way.  If [/dev/urandom] is available on
-   the host machine, it is used to provide a highly random initial
-   seed.  Otherwise, a less random seed is computed from system
-   parameters (current time, process IDs). *)
+(** Initialize the domain-local generator with a random seed chosen
+    in a system-dependent way.  If [/dev/urandom] is available on the host
+    machine, it is used to provide a highly random initial seed.  Otherwise, a
+    less random seed is computed from system parameters (current time, process
+    IDs). *)
```

Comment:
When /dev/urandom is not available, we have a randomness problem if many domains are spawned at the same time and initialize their PRNG right at the start. Is it possible to add the domain identifier to the mix?

Reply:
I've updated the the random state initialisation with a unique token of domain-local state.